power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -314,21 +293,21 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
         }
     }
 
-    // update all components
-    template <cache_type_c CacheType>
-    void update_component(ConstDataset const& update_data, Idx pos, SequenceIdxView const& sequence_idx_map) {
-        run_functor_with_all_types_return_void([this, pos, &update_data, &sequence_idx_map]<typename CT>() {
-            this->update_component<CT, CacheType>(update_data, pos, std::get<index_of_component<CT>>(sequence_idx_map));
-        });
-    }
-    template <cache_type_c CacheType>
-    void update_component(ConstDataset const& update_data, Idx pos, SequenceIdx const& sequence_idx_map) {
-        run_functor_with_all_types_return_void([this, pos, &update_data, &sequence_idx_map]<typename CT>() {
-            this->update_component<CT, CacheType>(update_data, pos, std::get<index_of_component<CT>>(sequence_idx_map));
-        });
+    // update all components across all scenarios
+    template <cache_type_c CacheType, typename SequenceIdxMap>
+        requires(std::same_as<SequenceIdxMap, SequenceIdx> || std::same_as<SequenceIdxMap, SequenceIdxView>)
+    void update_components(ConstDataset const& update_data, Idx pos, SequenceIdxMap const& sequence_idx_map) {
+        main_core::utils::run_functor_with_all_types_return_void<ComponentType...>(
+            [this, pos, &update_data, &sequence_idx_map]<typename CT>() {
+                this->update_component_row_col<CT, CacheType>(update_data, pos,
+                                                              std::get<index_of_component<CT>>(sequence_idx_map));
+            });
     }
-    template <cache_type_c CacheType> void update_component(ConstDataset const& update_data) {
-        update_component<CacheType>(update_data, 0, get_sequence_idx_map(update_data.get_individual_scenario(0)));
+    // update all components in the first scenario (e.g. permanent update)
+    template <cache_type_c CacheType> void update_components(ConstDataset const& update_data) {
+        auto const sequence_idx_map = main_core::update_independence::get_sequence_idx_map<ComponentType...>(
+            state_, update_data.get_individual_scenario(0));
+        update_components<CacheType>(update_data, 0, sequence_idx_map);
```

Comment:
```suggestion
    template <cache_type_c CacheType> void update_components(ConstDataset const& update_data) {
        auto const sequence_idx_map = main_core::update_independence::get_sequence_idx_map<ComponentType...>(
            state_, update_data);
        update_components<CacheType>(update_data, 0, sequence_idx_map);
```

Also i think renaming it to get_all_sequence_idx_map would make it more readable instead